src/Pages/CardReader/CardReader.js

```diff
@@ -1,28 +1,65 @@
 import { useState, useEffect } from 'react';
 import { BASE_API_URL } from '../../Enums';
+import { getCardsFromDb, deleteCardFromDb } from '../../APIFunctions/CardReader';
+import ConfirmationModal from '../../Components/DecisionModal/ConfirmationModal';
+import { trashcanSymbol } from '../Overview/SVG';
 
 export default function CardReader(props) {
   const [logs, setLogs] = useState([]);
-  const [error, setError] = useState('');
-  const token = props.user.token;
+  const [cards, setCards] = useState([]);
+  const [toggleDelete, setToggleDelete] = useState(false);
+  const [cardToDelete, setCardToDelete] = useState({});
+  const [tab, setTab] = useState(() => {
+    const params = new URLSearchParams(window.location.search);
+    return params.get('tab') || 'registry';
+  });
+  const [connected, setConnected] = useState(true);
 
   const buildLog = (data) => {
     let date = new Date().toISOString().padEnd(30, ' ');
     let endpoint = data.endpoint.padEnd(20, ' ');
-    let statusCode = String(data.statusCode).padEnd(8, ' ');
+    let statusCode = String(data.statusCode).padEnd(15, ' ');
     return [date, endpoint, statusCode, data.message].join('');
   };
 
+  const header = [
+    'TIMESTAMP'.padEnd(30),
+    'ENDPOINT'.padEnd(20),
+    'STATUS CODE'.padEnd(15),
+    'MESSAGE\n'
+  ].join('');
```

Comment:
to skip recalculating this on rerenders we can use useMemo to calcualte the header

https://react.dev/reference/react/useMemo